docs/syntax_and_semantics/compile_time_flags.md

```diff
@@ -1,48 +1,149 @@
 # Compile-time flags
 
-Types, methods and generally any part of your code can be conditionally defined based on some flags available at compile time. These flags are by default read from the hosts [LLVM Target Triple](http://llvm.org/docs/LangRef.html#target-triple), split on `-`. To get the target you can execute `llvm-config --host-target`.
+Compile-time flags are boolean values provided through the compiler via a macro method.
+They allow to conditionally include or exclude code based on compile time conditions.
 
-```console
-$ llvm-config --host-target
-x86_64-unknown-linux-gnu
-
-# so the flags are: x86_64, unknown, linux, gnu
-```
+There are several default flags provided by the compiler with information about compiler options and the target platform.
+User-provided flags are passed to the compiler, which allow them to be used as feature flags.
 
-To define a flag, simply use the `--define` or `-D` option, like so:
-
-```bash
-crystal some_program.cr -Dflag
-```
+## Querying flags
 
-Additionally, if a program is compiled with `--release`, the `release` flag will be set.
+A flag is just a named identifier which is either set or not.
+The status can be queried from code via the macro method [`flag?`](https://crystal-lang.org/api/1.2.1/Crystal/Macros.html#flag%3F%28name%29%3ABoolLiteral-instance-method). It receives the name of a flag as a string or symbol
+literal and returns a bool literal indicating the flag's state.
 
-You can check if a flag is defined with the `flag?` macro method:
+The following program shows the use of compile-time flags by printing the target OS family.
 
-```crystal
-{% if flag?(:x86_64) %}
-  # some specific code for 64 bits platforms
+```cr
+{% if flag?(:unix) %}
+  puts "Compiling for a UNIX-like operating system"
+{% elsif flag?(:windows) %}
+  puts "Compiling for Windows"
```

Comment:
It's confusing that it says "compiling" as the currently ongoing action but it says that *at runtime*

```suggestion
  puts "This program is compiled for a UNIX-like operating system"
{% elsif flag?(:windows) %}
  puts "This program is compiled for Windows"
```